src/EPR.SubsidiaryBulkUpload.Application/Services/BulkUploadOrchestration.cs

```diff
@@ -65,39 +66,73 @@ public async Task NotifyErrors(IEnumerable<CompaniesHouseCompany> data, UserRequ
 
         // parents companies house company number not found report
-        var subsidiaryGroupsAndParentOrgWith_InvalidCompaniesHouseNumber = subsidiaryGroupsAndParentOrg.Where(sg => sg.parentOrg != null && sg.SubsidiaryGroup.Parent.companies_house_number != sg.parentOrg.companiesHouseNumber).Select(s => s.SubsidiaryGroup);
+        var subsidiaryGroupsAndParentOrgWith_InvalidCompaniesHouseNumber = subsidiaryGroupsAndParentOrg.Where(sg => sg.ParentOrg != null && sg.SubsidiaryGroup.Parent.companies_house_number != sg.ParentOrg.companiesHouseNumber).Select(s => s.SubsidiaryGroup);
         await ReportCompanies(subsidiaryGroupsAndParentOrgWith_InvalidCompaniesHouseNumber.ToList(), userRequestModel, BulkUpdateErrors.ParentOrganisationFoundCompaniesHouseNumberNotMatchingMessage, BulkUpdateErrors.ParentOrganisationFoundCompaniesHouseNumberNotMatching);
 
-        var subsidiaryGroupsAndParentOrgWithValidCompaniesHouseNumber = subsidiaryGroupsAndParentOrg.Where(sg => sg.parentOrg != null && sg.SubsidiaryGroup.Parent.companies_house_number == sg.parentOrg.companiesHouseNumber);
+        var subsidiaryGroupsAndParentOrgWithValidCompaniesHouseNumber = subsidiaryGroupsAndParentOrg.Where(sg => sg.ParentOrg != null && sg.SubsidiaryGroup.Parent.companies_house_number == sg.ParentOrg.companiesHouseNumber);
 
         var addedSubsidiariesCount = 0;
 
         foreach (var subsidiaryGroupAndParentOrg in subsidiaryGroupsAndParentOrgWithValidCompaniesHouseNumber)
         {
-            addedSubsidiariesCount += await childProcessor.Process(
-                subsidiaryGroupAndParentOrg.SubsidiaryGroup.Subsidiaries,
+            var subsidiariesToProcess = await FilterDuplicateSubsidiaries(userRequestModel, subsidiaryGroupAndParentOrg);
+
+            addedSubsidiariesCount += await _childProcessor.Process(
+                subsidiariesToProcess,
                 subsidiaryGroupAndParentOrg.SubsidiaryGroup.Parent,
-                subsidiaryGroupAndParentOrg.parentOrg,
+                subsidiaryGroupAndParentOrg.ParentOrg,
                 userRequestModel);
         }
 
         await _notificationService.SetStatus(userRequestModel.GenerateKey(NotificationStatusKeys.SubsidiaryBulkUploadProgress), "Finished");
         await _notificationService.SetStatus(userRequestModel.GenerateKey(NotificationStatusKeys.SubsidiaryBulkUploadRowsAdded), addedSubsidiariesCount.ToString());
     }
 
+    private async Task<List<CompaniesHouseCompany>> FilterDuplicateSubsidiaries(
+        UserRequestModel userRequestModel,
+        (ParentAndSubsidiaries SubsidiaryGroup, OrganisationResponseModel? ParentOrg) subsidiaryGroupAndParentOrg)
+    {
+        var duplicatesGroup = subsidiaryGroupAndParentOrg.SubsidiaryGroup.Subsidiaries.GroupBy(companiesHouseCompany => new
+        {
+            companiesHouseCompany.subsidiary_id,
+            companiesHouseCompany.organisation_name,
+            companiesHouseCompany.companies_house_number,
```

Comment:
@kaderhashmieviden  we have discussed this block.
core group id is Companies house number, additional are org name and we have subsidiary id added now. rest is not required. as explained child items already filtered out before reaching this iteration

Reply:
This code identifies the duplicates using all 3 columns in a group. The duplicates are not being filtered before this point (happy to demo on my machine using your example file).